Assets/MixedRealityToolkit/Providers/BaseDeviceManager.cs

```diff
@@ -67,5 +67,7 @@ protected virtual IMixedRealityPointer[] RequestPointers(SystemType controllerTy
 
             return pointers.Count == 0 ? null : pointers.ToArray();
         }
+
+        public virtual void PreServiceUpdate() { }
```

Comment:
I think something like `BeforeUpdate` or something similar?

The update is driven by Unity's game loop events anyway. They're not really specific to services.

Anyone have thoughts on a name?

Reply:
`EarlyUpdate` exists in Unity's experimental API.